app/models/tenant.rb

```diff
@@ -300,6 +300,7 @@ def validate_only_one_root
   end
 
   def create_tenant_group
+    reload
```

Comment:
@mkanoor tested this with an `after_commit :create_tenant_group, :on => :create` and it worked

@matthewd I'd prefer that solution - thoughts?